packages/numbers/lib/numbers.ts

```diff
@@ -20,45 +25,93 @@ const runRequest = async <T>(options: NumbersClassParameters): Promise<NumbersRe
+        if (oldObject[mapping[key]]) {
+            newObject[key] = oldObject[mapping[key]];
+            delete oldObject[mapping[key]]
+        }
+    }
+    newObject = { ...newObject, ...oldObject };
+    return newObject;
+}
+
 const BASE_URL = "https://rest.nexmo.com".replace(/\/+$/, "");
 
 export const NumbersParamCreator = function (options?: NumbersClassParameters) {
     return {
         buyNumber(params?: NumbersParams) {
+            let mapping = { 'target_api_key': 'targetApiKey' };
+            let data: NumbersQueryParams = {
+                country: params.country,
+                msisdn: params.msisdn
+            };
+            data = remapObjects(mapping, data, params);
+
             const localVetchOptions = {};
             localVetchOptions['url'] = `${options.baseUrl}/number/buy`;
             localVetchOptions['headers'] = Object.assign({}, options.headers);
-            localVetchOptions['data'] = options.auth.getQueryParams(params);
+            localVetchOptions['data'] = options.auth.getQueryParams(data);
             localVetchOptions['method'] = 'POST';
             return localVetchOptions;
         },
         cancelNumber(params?: NumbersParams) {
+            let mapping = { 'target_api_key': 'targetApiKey' };
+            let data: NumbersQueryParams = {
+                country: params.country,
+                msisdn: params.msisdn
+            };
+            data = remapObjects(mapping, data, params);
+
             const localVetchOptions = {};
             localVetchOptions['url'] = `${options.baseUrl}/number/cancel`;
             localVetchOptions['headers'] = Object.assign({}, options.headers);
-            localVetchOptions['data'] = options.auth.getQueryParams(params);
+            localVetchOptions['data'] = options.auth.getQueryParams(data);
             localVetchOptions['method'] = 'POST';
             return localVetchOptions;
         },
         getAvailableNumbers(filter?: NumbersSearchFilter) {
+            let mapping = {
+                "search_pattern": "searchPattern"
+            };
+            let data: NumbersQuerySearchFilter = { country: filter.country };
+            data = remapObjects(mapping, data, filter);
+
             const localVetchOptions = {};
             localVetchOptions['url'] = `${options.baseUrl}/number/search`;
             localVetchOptions['headers'] = Object.assign({}, options.headers);
-            localVetchOptions['params'] = options.auth.getQueryParams(filter);
+            localVetchOptions['params'] = options.auth.getQueryParams(data);
             return localVetchOptions;
         },
         getOwnedNumbers(filter?: NumbersOwnedFilter) {
+            let mapping = {
+                'application_id': 'applicationId',
+                'has_application': 'hasApplication',
+                'search_pattern': 'searchPattern',
+            }
+            let data: NumbersQueryOwnedFilter = {};
+            data = remapObjects(mapping, data, filter);
```

Comment:
same issue as above of setting and passing same variable
```suggestion
            data = remapObjects(mapping, data, filter);
```